domain/model/mesh_results.py

```diff
@@ -1,21 +1,57 @@
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from datetime import datetime, timezone
-from typing import List, Optional
+from typing import Dict, List, Optional, Tuple
 
+from domain.geo import Coordinates
+from domain.metric_type import MetricType
 from domain.types import AgentID, MetricValue
 
 
+@dataclass
+class Agent:
+    id: AgentID = AgentID()
+    ip: str = ""
+    name: str = ""
+    alias: str = ""
+    coords: Coordinates = Coordinates()
+
+
+class Agents:
+    def __init__(self) -> None:
+        self._agents: Dict[AgentID, Agent] = {}
+
+    def get_by_id(self, id: AgentID) -> Agent:
+        if id in self._agents:
+            return self._agents[id]
+        return Agent()
+
+    def get_by_alias(self, alias: str) -> Agent:
+        for _, v in self._agents.items():
+            if v.alias == alias:
+                return v
+        return Agent()
+
+    def get_alias(self, id: AgentID) -> str:
+        agent = self.get_by_id(id)
+        if agent.id == AgentID():
```

Comment:
The test against a specific instance IMHO obfuscates the code. Returning `None` from `get_by_id` would be more in line with POLA ([Principle Of Least Astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)).